src/concerns/focus.coffee

```diff
@@ -0,0 +1,28 @@
+###
+Code related to focus and hover state
+###
+export default
+
+	# Some simple data about our component and window its mounted on
+	data: ->
+		hovered: false
+		buttonFocused: false
+		windowVisible: true
+
+	computed:
+		isFocused: -> @windowVisible and @hovered
+		windowHidden: -> !@windowVisible
+
+	methods:
+		onEnter: -> @hovered = true
+		onLeave: -> @hovered = false
+
+		# Updates @windowVisible based on our document
+		updateVisibility: -> @windowVisible = !document.hidden
+
+	# Watch the visibility updates of our document
+	mounted: ->
+		document.addEventListener 'visibilitychange', @updateVisibility
```

Comment:
To make most performant, I think we should only add these listeners if we need them.  Check out in vue-visual how I check (via a computed property) whether I need IntersectionObserver before I add it: 

https://github.com/BKWLD/vue-visual/blob/a149d07ccd51036e717e6ab874fa725b5fe8285e/concerns/observes-viewport.coffee#L34

Reply:
Will do. I was hesitant to do this at first in case we added more functionality that wanted to use the `focus.vue` concerns, but think that `watchesHover` computed property will help solve that concern.